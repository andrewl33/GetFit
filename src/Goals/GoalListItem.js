import React, { PureComponent } from 'react';
import { StyleSheet, Text, View } from 'react-native';
import PropTypes from 'prop-types';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 10,
    borderRadius: 2,

    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: 20,
  },
});

export default class GoalListItem extends PureComponent {
  render() {
    const { title } = this.props;

    return (
      <View style={styles.container}>
        <Text style={styles.text}>{title}</Text>
      </View>
    );
  }
}

// defines what is passed down to GoalListItem
GoalListItem.propTypes = {
  title: PropTypes.string.isRequired,
};
