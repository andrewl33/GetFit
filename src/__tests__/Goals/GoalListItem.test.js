import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import GoalListItem from '../../Goals/GoalListItem';

const item = {
  id: 2,
  title: 'Goal Item 2',
  createDate: '1-10-2001',
  endDate: '1-15-2001',
  isActive: true,
  unitType: 'miles',
  startAmount: 0,
  progress: 2.5,
  endAmount: 10,
};

describe('GoalListItem', () => {
  it('renders', () => {
    const tree = renderer.create(<GoalListItem {...item} />);
    expect(tree).toMatchSnapshot();
  });
});
