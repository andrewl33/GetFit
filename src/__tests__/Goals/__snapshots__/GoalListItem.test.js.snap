// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GoalListItem renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "borderRadius": 2,
      "flex": 1,
      "justifyContent": "center",
      "padding": 10,
    }
  }
>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Goal Item 2
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Start:
    1-10-2001
     End:
    1-15-2001
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    Progress: 
    2.5
     
    miles
     out of 
    10
     
    miles
  </Text>
  <Text
    style={
      Object {
        "fontSize": 20,
      }
    }
  >
    7.5
     
    miles
     to go!
  </Text>
</View>
`;
